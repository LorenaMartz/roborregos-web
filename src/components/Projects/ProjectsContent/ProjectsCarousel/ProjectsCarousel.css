.project-carousels {
    height: auto;
}

.project-carousel {
    background-color: black;
    height: 90vh;
    border-bottom: 0.5vh solid white;
}