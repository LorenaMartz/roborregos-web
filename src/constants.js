export const LARGE_WIDTH = 1200
export const MEDIUM_WIDTH = 992
export const SMALL_WIDTH = 768
export const XSMALL_WIDTH = 592
export const MOBILE_WIDTH = 414
