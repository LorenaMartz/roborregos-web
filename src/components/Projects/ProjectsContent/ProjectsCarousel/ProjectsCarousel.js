// @flow
import React, { Component } from 'react'
import Carousel from 'react-material-ui-carousel'
import ProjectsCard from '../ProjectsCard/ProjectsCard'
import './ProjectsCarousel.css'

type Project = {
  title: string,
  description: string,
  image: string,
  wiki: string
};

type Props = {
  project_carousels: Array<Array<Project>>,
  main_counter: number
};

/** Component class of Projects page. */
class ProjectsContent extends Component <Props> {
  project_carousels: Array<Array<Project>>;

  main_counter: number;

  /**
   * Class constructor
   * @param {list} props: List of projects data.
   */
  constructor(props: Props) {
    super(props)
    this.project_carousels = props.project_carousels
    this.main_counter = props.main_counter
  }

  joinUsCallback = (link: string) => {
    window.open(link)
  }

  /**
   * Renders Responsive view of Projects's carousels.
   * @return {renderized_component} Carrousels with project cards on them.
   */
  render() {
    return (
      <div className="project-carousels">
        {
            this.project_carousels.map((projects, carrousel_index) => (
              <Carousel
                className="project-carousel"
                navButtonsAlwaysVisible
                timeout={200}
                animation="slide"
                autoPlay={false}
                index={(((carrousel_index % 2) + (this.main_counter % 2))) % 2}
              >
                {projects.map((project, index) => (
                  <ProjectsCard
                    project={project}
                    index={index}
                    show_scrollers={false}
                  />
                ))}
              </Carousel>
            ))
        }
      </div>
    )
  }
}

export default ProjectsContent
