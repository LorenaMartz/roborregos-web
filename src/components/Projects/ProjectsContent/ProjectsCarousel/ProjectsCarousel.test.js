// @flow
import React from 'react'
import { shallow, configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ProjectsCarousel from './ProjectsCarousel'
import { LARGE_WIDTH, SMALL_WIDTH } from '../../../../constants'

let wrapper
configure({ adapter: new Adapter() })

const carousels = [
  [
    {
      title: 'main-project-1',
      description: 'description-main-1',
      wiki: 'https://github.com/1',
      image: 'main-project-1',
    },
    {
      title: 'main-project-2',
      description: 'description-main-2',
      wiki: 'https://github.com/2',
      image: 'main-project-2',
    },
  ],
  [
    {
      title: 'main-project-3',
      description: 'description-main-3',
      wiki: 'https://github.com/3',
      image: 'main-project-3',
    },
    {
      title: 'main-project-4',
      description: 'description-main-4',
      wiki: 'https://github.com/4',
      image: 'main-project-4',
    },
  ],
]

const main_projects_length = 3

beforeEach(() => {
  wrapper = shallow(
    <ProjectsCarousel project_carousels={carousels} main_counter={main_projects_length} />,
  )
})

it('<ProjectsCarousel> renders large view components correctly', () => {
  expect(wrapper.find('.project-carousels').length).toBe(1)
  expect(wrapper.find('.project-carousel').length).toBe(2)
})

// Snapshots
describe('<ProjectsCarousel> matches snapshot with:', () => {
  it('Large width', () => {
    global.innerWidth = LARGE_WIDTH
    global.dispatchEvent(new Event('resize'))
    const wrapper_content = mount(
      <ProjectsCarousel project_carousels={carousels} main_counter={main_projects_length} />,
    ).debug()
    expect(wrapper_content).toMatchSnapshot('ProjectsCarouselLargeView.test.js.snap')
  })

  it('Small width', () => {
    global.innerWidth = SMALL_WIDTH
    global.dispatchEvent(new Event('resize'))
    const wrapper_content = mount(
      <ProjectsCarousel project_carousels={carousels} main_counter={main_projects_length} />,
    ).debug()
    expect(wrapper_content).toMatchSnapshot('ProjectsCarouselSmallView.test.js.snap')
  })
})
