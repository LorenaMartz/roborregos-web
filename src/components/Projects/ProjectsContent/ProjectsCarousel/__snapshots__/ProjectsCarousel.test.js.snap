// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectsCarousel> matches snapshot with: Large width: ProjectsCarouselLargeView.test.js.snap 1`] = `
"<div className=\\"project-carousels\\">
  <WithStyles(Carousel) className=\\"project-carousel\\" navButtonsAlwaysVisible={true} timeout={200} animation=\\"slide\\" autoPlay={false} index={1}>
    <ProjectsCard project={{...}} index={0} show_scrollers={false} />
    <ProjectsCard project={{...}} index={1} show_scrollers={false} />
  </WithStyles(Carousel)>
  <WithStyles(Carousel) className=\\"project-carousel\\" navButtonsAlwaysVisible={true} timeout={200} animation=\\"slide\\" autoPlay={false} index={0}>
    <ProjectsCard project={{...}} index={0} show_scrollers={false} />
    <ProjectsCard project={{...}} index={1} show_scrollers={false} />
  </WithStyles(Carousel)>
</div>"
`;

exports[`<ProjectsCarousel> matches snapshot with: Small width: ProjectsCarouselSmallView.test.js.snap 1`] = `
"<div className=\\"project-carousels\\">
  <WithStyles(Carousel) className=\\"project-carousel\\" navButtonsAlwaysVisible={true} timeout={200} animation=\\"slide\\" autoPlay={false} index={1}>
    <ProjectsCard project={{...}} index={0} show_scrollers={false} />
    <ProjectsCard project={{...}} index={1} show_scrollers={false} />
  </WithStyles(Carousel)>
  <WithStyles(Carousel) className=\\"project-carousel\\" navButtonsAlwaysVisible={true} timeout={200} animation=\\"slide\\" autoPlay={false} index={0}>
    <ProjectsCard project={{...}} index={0} show_scrollers={false} />
    <ProjectsCard project={{...}} index={1} show_scrollers={false} />
  </WithStyles(Carousel)>
</div>"
`;
